internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -246,8 +246,13 @@ func (s *Server) Rollback(ctx context.Context, in *machine.RollbackRequest) (*ma
 			}
 		}()
 
+		disk, err := s.Controller.Runtime().Config().Machine().Install().Disk()
+		if err != nil {
+			return err
+		}
+
 		grub := &grub.Grub{
-			BootDisk: s.Controller.Runtime().Config().Machine().Install().Disk(),
```

Comment:
feels like this was wrong from the beginning - it should be using the disk Talos is installed to, and we had that method somewhere in runtime/state?

Reply:
should be fixed now